src/application/controller/gain_scheduling.c

```diff
@@ -1 +1,3471 @@
+     -4.2542E-01, -4.2850E-01, -4.3195E-01, -4.3586E-01, -4.4034E-01, -4.4558E-01, -4.5181E-01,
+     -4.5942E-01, -4.6904E-01, -4.8176E-01, -4.9979E-01, -5.2838E-01, -5.8456E-01, -7.9267E-01,
+     7.9267E-01,  5.8456E-01,  5.2838E-01,  4.9979E-01,  4.8176E-01,  4.6904E-01,  4.5942E-01,
+     4.5181E-01,  4.4558E-01,  4.4034E-01,  4.3586E-01,  4.3195E-01,  4.2850E-01,  4.2542E-01,
+     4.2264E-01,  -4.1335E-01, -4.1605E-01, -4.1904E-01, -4.2239E-01, -4.2619E-01, -4.3055E-01,
+     -4.3563E-01, -4.4168E-01, -4.4906E-01, -4.5838E-01, -4.7071E-01, -4.8818E-01, -5.1586E-01,
+     -5.7021E-01, -7.7131E-01, 7.7131E-01,  5.7021E-01,  5.1586E-01,  4.8818E-01,  4.7071E-01,
+     4.5838E-01,  4.4906E-01,  4.4168E-01,  4.3563E-01,  4.3055E-01,  4.2619E-01,  4.2239E-01,
+     4.1904E-01,  4.1605E-01,  4.1335E-01,  -4.0462E-01, -4.0725E-01, -4.1016E-01, -4.1342E-01,
+     -4.1711E-01, -4.2135E-01, -4.2628E-01, -4.3216E-01, -4.3933E-01, -4.4838E-01, -4.6034E-01,
+     -4.7729E-01, -5.0412E-01, -5.5678E-01, -7.5137E-01, 7.5137E-01,  5.5678E-01,  5.0412E-01,
+     4.7729E-01,  4.6034E-01,  4.4838E-01,  4.3933E-01,  4.3216E-01,  4.2628E-01,  4.2135E-01,
+     4.1711E-01,  4.1342E-01,  4.1016E-01,  4.0725E-01,  4.0462E-01,  -3.9640E-01, -3.9896E-01,
+     -4.0179E-01, -4.0497E-01, -4.0856E-01, -4.1268E-01, -4.1749E-01, -4.2320E-01, -4.3017E-01,
+     -4.3896E-01, -4.5059E-01, -4.6705E-01, -4.9310E-01, -5.4418E-01, -7.3270E-01, 7.3270E-01,
+     5.4418E-01,  4.9310E-01,  4.6705E-01,  4.5059E-01,  4.3896E-01,  4.3017E-01,  4.2320E-01,
+     4.1749E-01,  4.1268E-01,  4.0856E-01,  4.0497E-01,  4.0179E-01,  3.9896E-01,  3.9640E-01,
+     -3.8864E-01, -3.9113E-01, -3.9390E-01, -3.9699E-01, -4.0049E-01, -4.0451E-01, -4.0919E-01,
+     -4.1475E-01, -4.2153E-01, -4.3009E-01, -4.4139E-01, -4.5740E-01, -4.8271E-01, -5.3231E-01,
+     -7.1518E-01, 7.1518E-01,  5.3231E-01,  4.8271E-01,  4.5740E-01,  4.4139E-01,  4.3009E-01,
+     4.2153E-01,  4.1475E-01,  4.0919E-01,  4.0451E-01,  4.0049E-01,  3.9699E-01,  3.9390E-01,
+     3.9113E-01,  3.8864E-01,  -3.8130E-01, -3.8373E-01, -3.8643E-01, -3.8945E-01, -3.9286E-01,
+     -3.9678E-01, -4.0134E-01, -4.0676E-01, -4.1337E-01, -4.2170E-01, -4.3271E-01, -4.4828E-01,
+     -4.7290E-01, -5.2112E-01, -6.9870E-01, 6.9870E-01,  5.2112E-01,  4.7290E-01,  4.4828E-01,
+     4.3271E-01,  4.2170E-01,  4.1337E-01,  4.0676E-01,  4.0134E-01,  3.9678E-01,  3.9286E-01,
+     3.8945E-01,  3.8643E-01,  3.8373E-01,  3.8130E-01,  -3.7434E-01, -3.7672E-01, -3.7935E-01,
+     -3.8230E-01, -3.8563E-01, -3.8945E-01, -3.9390E-01, -3.9919E-01, -4.0563E-01, -4.1376E-01,
+     -4.2448E-01, -4.3965E-01, -4.6363E-01, -5.1055E-01, -6.8316E-01, 6.8316E-01,  5.1055E-01,
+     4.6363E-01,  4.3965E-01,  4.2448E-01,  4.1376E-01,  4.0563E-01,  3.9919E-01,  3.9390E-01,
+     3.8945E-01,  3.8563E-01,  3.8230E-01,  3.7935E-01,  3.7672E-01,  3.7434E-01,  -3.6773E-01,
+     -3.7006E-01, -3.7263E-01, -3.7551E-01, -3.7877E-01, -3.8250E-01, -3.8684E-01, -3.92
```

Comment:
```suggestion
static arm_bilinear_interp_instance_f32 *gain_instance_arr[GAIN_NUM] = {
{ .numRows = GAIN_P_SIZE, .numCols = GAIN_C_SIZE, .pData = &gain_table[0][0] },
{ .numRows = GAIN_P_SIZE, .numCols = GAIN_C_SIZE, .pData = &gain_table[1][0] },
{ .numRows = GAIN_P_SIZE, .numCols = GAIN_C_SIZE, .pData = &gain_table[2][0] },
{ .numRows = GAIN_P_SIZE, .numCols = GAIN_C_SIZE, .pData = &gain_table[3][0] },
};
```
I suppose this defeats the purpose of having `GAIN_NUM` be dynamic, but I would like to avoid adding functions where possible

Reply:
I had this at first lol then changed it lol